### ISSUE TITLE
ERROR:  function round(double precision, integer) does not exist

### ISSUE THREAD

**Issue by strk** [qgis/QGIS#31495]
With current master, the postgresql tests fail with this error (among others):
```
====================================================================
PostGIS(1): Query: DECLARE qgis_1166 BINARY CURSOR FOR SELECT "key1"::text,"key2"::text,"pk"::text,"cnt"::text,"name"::text,"name2"::text,"num_char"::text FROM "qgis_test"."someDataCompound" WHERE (round(("cnt" / ((66.67)::real)),0) <= 2) returned 7 [ERROR:  function round(double precision, integer) does not exist                                        ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
]
PostGIS(1): 1 cursor states lost.
SQL: DECLARE qgis_1166 BINARY CURSOR FOR SELECT "key1"::text,"key2"::text,"pk"::text,"cnt"::text,"name"::text,"name2"::text,"num_char"::text FROM "qgis_test"."someDataCompound" WHERE (round(("cnt" / ((66.67)::real)),0) <= 2)
LINE 1: ...::text FROM "qgis_test"."someDataCompound" WHERE (round(("cn...
                                                    ==
FAIL: testStyleDatabaseWithService (__main__.TestPyQgsPostgresProvider)
Test saving style in DB using a service file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/qgis/qgis-master/tests/src/python/test_provider_postgres.py", line 1318, in testStyleDatabaseWithService
    self.assertEqual(len(ids), 1)
AssertionError: 0 != 1

```

I'm running it against PostgreSQL-9.6.9
Indeed round(float8, int) does not exist, while round(numeric, int) does:
```
strk=# select round(1.01::float8, 2);
ERROR:  function round(double precision, integer) does not exist
LINE 1: select round(1.01::float8, 2);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
strk=# select round(1.01::numeric, 2);
 round
-------
  1.01
(1 row)

strk=# select version();
                                             version                                              
--------------------------------------------------------------------------------------------------
 PostgreSQL 9.6.9 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0, 64-bit
(1 row)
```

**Comment by gioman** [comment]
@strk is this still relevant?

**Comment by MorriganR** [comment]
fixed: #42789
